streams/src/main/java/org/apache/kafka/streams/state/internals/ValueAndTimestampSerde.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+
+import java.util.Map;
+
+public class ValueAndTimestampSerde<V> implements Serde<ValueAndTimestamp<V>> {
+    private ValueAndTimestampSerializer<V> valueAndTimestampSerializer;
+    private ValueAndTimestampDeserializer<V> valueAndTimestampDeserializer;
+    private boolean initialized = false;
```

Comment:
I'm not fully understanding the boolean flag here: when would we call the constructor without a `valueSerde` and call `init(valueSerde)` later? I'm thinking:

1) If users pass in a serde via `Materialized` or via `Stores`, Streams could enforce them to always have the `valueSerde` passed in when constructing the valueAndTimestampSerde.
2) If user does not specify a serde, then Streams code internally could just always construct the valueAndTimestampSerde by passing the default serde as `valueSerde` when constructing it (i.e. in `MeteredTimestampedKeyValueStore` we can omit checking `initialized`).

Reply:
Ack. (I guess this is some left over after code refactoring that slipped.)